src/ui/src/core/index.ts

```
@@ -699,6 +711,30 @@ export function generateCore() {
 	}
 
+	function waitSyncHealthToBe(value: SyncHealth) {
+		if (syncHealth.value === value) return;
+
+		// eslint-disable-next-line no-async-promise-executor
+		return new Promise<void>(async (res) => {
+			while (syncHealth.value !== value) {
+				await new Promise((r) => setInterval(r, 1_000));
+			}
+			return res();
+		});
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)